clients/src/main/java/org/apache/kafka/clients/InFlightRequests.java

```diff
@@ -80,7 +87,9 @@ public void add(NetworkClient.InFlightRequest request) {
      * @return The request
      */
     public NetworkClient.InFlightRequest completeLastSent(String node) {
-        return requestQueue(node).pollFirst();
+        NetworkClient.InFlightRequest inFlightRequest = requestQueue(node).pollFirst();
```

Comment:
Same here, it is probably safer to only decrement counter when the inFlightRequest is not null.